### PR TITLE
fix(platform-browser): provide Title service as part of the module

### LINKED ISSUE #11600: TitleService is not export in RC.6
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior** 
In a shared module (SharedModule) I import and export BrowserModule. But other modules that use this shared module cannot find the provider for Title. If I add Title to SharedModule providers it does work in other modules.

**Expected behavior**
If I understand correctly I would expect that BrowserModule to export 'Title' for us.

**Please tell us about your environment:**
Mac OS X 10.11.6, angular-cli, ng serve

* **Angular version:** 2.0.0-rc.6
Does not appear in the fixed list for rc.7

* **Language:** Typescript 2.0

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Fixes #11600

### CODE REVIEW

File: modules/@angular/platform-browser/src/browser.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ import {WebAnimationsDriver} from '../src/dom/web_animations_driver';
 import {BrowserDomAdapter} from './browser/browser_adapter';
 import {BrowserPlatformLocation} from './browser/location/browser_platform_location';
 import {BrowserGetTestability} from './browser/testability';
+import {Title} from './browser/title';
```
Comment (vicb) [inline modules/@angular/platform-browser/src/browser.ts]: is `Title` (TS) exported ?

Comment (pkozlowski-opensource) [inline modules/@angular/platform-browser/src/browser.ts]: @vicb yes, it is, see https://github.com/angular/angular/blob/master/modules/%40angular/platform-browser/src/platform-browser.ts#L10

**Comment by vicb** [comment]
Would you mind adding a quick unit test ?

**Comment by pkozlowski-opensource** [comment]
@vicb @IgorMinar test added so removed the cleanup label. While not terribly critical it would be good to land this as part of final IMO. Shell I stick the merge label on it?

File: modules/@angular/platform-browser/test/browser/title_spec.ts [hunk 2]
```diff
@@ -29,6 +31,20 @@ export function main() {
...
+    class DependsOnTitle {
+      constructor(public title: Title) {}
+    }
+
+    beforeEach(() => {
+      TestBed.configureTestingModule({imports: [BrowserModule], providers: [DependsOnTitle]});
+    });
 
+    it('should inject Title service when using BrowserModule',
+       () => { expect(TestBed.get(DependsOnTitle).title).not.toBeNull(); });
```
Comment (vicb) [inline modules/@angular/platform-browser/test/browser/title_spec.ts]: toBeAnInstanceOf

File: modules/@angular/platform-browser/test/browser/title_spec.ts [hunk 3]
```diff
@@ -29,6 +31,20 @@ export function main() {
...
+
+  describe('integration test', () => {
+
+    @Injectable()
+    class DependsOnTitle {
+      constructor(public title: Title) {}
+    }
+
+    beforeEach(() => {
+      TestBed.configureTestingModule({imports: [BrowserModule], providers: [DependsOnTitle]});
```
Comment (vicb) [inline modules/@angular/platform-browser/test/browser/title_spec.ts]:
```
TestBed.configureTestingModule({
  imports: [BrowserModule], 
  providers: [DependsOnTitle],
});
```

trailing comma is important for clang

Comment (pkozlowski-opensource) [inline modules/@angular/platform-browser/test/browser/title_spec.ts]: Will change but this makes me feel dirty... Does it really matter?

**Review by vicb — APPROVED** [review]

**Pull request merged**